### PR TITLE
wallet: auto-freeze small unconfirmed UTXOs

### PR DESCRIPTION
**Pull request by SomberNight**
see #6960

this implements option (1) from https://github.com/spesmilo/electrum/issues/6960#issuecomment-765407023

### CODE REVIEW

File: electrum/wallet.py [hunk 1]
```diff
@@ -1323,7 +1326,46 @@ def is_frozen_address(self, addr: str) -> bool:
...
+        """
+        # confirmed UTXOs are fine; check this first for performance:
+        block_height = utxo.block_height
+        assert block_height is not None
+        if block_height > 0:
+            return False
+        # exempt large value UTXOs
+        value_sats = utxo.value_sats()
+        assert value_sats is not None
+        threshold = self.config.get('auto_freeze_unconf_utxo_value_threshold_sats', 5_000)
```
Comment (ecdsa) [inline electrum/wallet.py:1356]: That is quite a long variable name. Also, the "u" in utxo already means unconfirmed.
How about 'utxo_freeze_threshold'?

File: electrum/wallet.py [hunk 2]
```diff
@@ -1323,7 +1326,46 @@ def is_frozen_address(self, addr: str) -> bool:
...
-        return prevout_str in self._frozen_coins
+        frozen = self._frozen_coins.get(prevout_str, None)
+        # note: there are three possible states for 'frozen':
+        #       True/False if the user explicitly set it,
+        #       None otherwise
+        if frozen is None:
+            return self._is_coin_small_and_untrusted(utxo)
+        return bool(frozen)
+
+    def _is_coin_small_and_untrusted(self, utxo: PartialTxInput) -> bool:
```
Comment (ecdsa) [inline electrum/wallet.py:1337]: I guess you meant small_and_unconfirmed

**Comment by ecdsa** [comment]
merged manually

File: electrum/wallet.py (hunk 1 above)
Comment (SomberNight) [reply electrum/wallet.py:1356]: sure that's fine

File: electrum/wallet.py (hunk 2 above)
Comment (SomberNight) [reply electrum/wallet.py:1337]: I simply was not sure how to call it; either is fine.

**Pull request closed without merging**